### ISSUE TITLE
Automatic Background Removal technology

### ISSUE THREAD

**Issue by InternetMaster1** [pytorch/vision#1665]
I am looking for a deep learning library/sdk which can be used to remove the background from any image automatically (with quality as good as www.remove.bg).

I tried some image segmentation SDKs with pre-trained models such as Tensorflow Lite & Fritz AI, but the accuracy of the cutout mask was very low, amongst other issues.

Criteria :-

1) Background Removal rather than just Human/Portrait Segmentation

If the foreground consists of person holding a balloon, sittting on a chair, with a pet on his side, then I want all of this to get extracted. Not just the human cutout. The segmentation SDKs I tried are only extracting humans (the chair gets vanished), that too with a very low quality mask (hair gets cut, parts of ear gets cut, etc).

2) Mask quality should be Super-Accurate

I want even the finer details like the hair, delicate clothes, etc to be extracted perfectly.

3) Fast & Lightweight (for mobile phone)

I want to use this technology on mobile phones (in an Android app) which should ideally work even in an offline environment. If this option is difficult to achieve, then plan B would be install the technoloy on our server.

4) Technology
What technology should I be exploring to achieve this? Is it called image segmentation or the better term would be image matting? (e.g. http://alphamatting.com/eval_25.php)

I have been reading a lot and I am currently lost in the sea of various technologies out there (OpenCV, Deep Matting, Mask RCNN, Instance Segmentation, Detectron2, Tensorflow, Pytorch, etc). I wonder what magic is happening behind the curtains of www.remove.bg

Would your library help me me to achieve what I am looking for? Any help you could provide would be awesome.

Thanks a ton!

**Comment by fmassa** [comment]
Hi,

You could try using `DeepLabV3` or Mask R-CNN, both are implemented in torchvision.
Check https://pytorch.org/blog/torchvision03/ to see if you'd need one model type or the other.

Regardless of the model used, you might probably need to test them to see if they fit your needs, but most probably you'll need to combine them with other techniques for getting the fine-grained details to work.

**Comment by InternetMaster1** [comment]
Could you advice about the additional techniques that might help with the
fine grained details. That is a struggle for me.

Thanks.

**Comment by fmassa** [comment]
Approaches leveraging alpha mating, for example https://arxiv.org/pdf/1703.03872.pdf or http://xiaoyongshen.me/papers/deepmatting.pdf

**Comment by InternetMaster1** [comment]
Are you aware of any Github libraries which are providing alpha matting ?

**Comment by fmassa** [comment]
I'm not aware, but a Google search will probably give you a lot of results

**Comment by InternetMaster1** [comment]
Thanks :)
